v2/internal/testcommon/samples_tester.go

```diff
@@ -228,3 +235,43 @@ func IsSampleExcluded(path string, exclusions []string) bool {
 	}
 	return false
 }
+
+func (t *SamplesTester) findAndSetARMIDReferences(obj genruntime.ARMMetaObject) (genruntime.ARMMetaObject, error) {
+	resourceRefType := reflect.TypeOf(genruntime.ResourceReference{})
+	visitor := reflecthelpers.NewReflectVisitor()
+	visitor.VisitStruct = func(this *reflecthelpers.ReflectVisitor, it reflect.Value, ctx interface{}) error {
+
+		if it.Type() == resourceRefType {
+			if it.CanInterface() {
+				reference := it.Interface().(genruntime.ResourceReference)
+				if reference.ARMID != "" {
+					armIDField := it.FieldByName("ARMID")
+					if !armIDField.CanSet() {
+						return errors.New("cannot set 'ARMID' field of 'genruntime.ResourceReference'")
+					}
+
+					//Regex to match '/00000000-0000-0000-0000-000000000000/' strings, to replace with the subscriptionID
+					subMatcher := regexp.MustCompile("\\/([0]+-?)+\\/")
+					armIDString := armIDField.String()
+					armIDString = strings.ReplaceAll(armIDString, defaultResourceGroup, t.rgName)
+					armIDString = subMatcher.ReplaceAllString(armIDString, fmt.Sprint("/", t.azureSubscription, "/"))
+
+					armIDField.SetString(armIDString)
+				}
+			} else {
+				// This should be impossible given how the visitor works
+				return errors.New("genruntime.ResourceReference field was unexpectedly nil")
+			}
+			return nil
+		}
+
+		return reflecthelpers.IdentityVisitStruct(this, it, ctx)
+	}
```

Comment:
Suggest pulling this out as a named function (can be a member of `SamplesTester`) - not only will that make `findAndSetARMIDReferences` easier to read, but the name will be a good clue about what the function does.